### PR TITLE
Make GHA workflow to retrieve latest promote-able SHA from master

### PR DESCRIPTION
**Pull request by swang392**
Relates to #76700

**Overview:** Wrote GHA to get the latest commit SHA. Another component of the script is pushing this SHA to the viable/strict branch but I'm planning to test that locally after verifying that this part is correct. 

**Test Plan:** Monitor github actions results to see if the SHA printed is correct -- I wasn't able to check this on my personal fork.

### CODE REVIEW

File: .github/workflows/update-viablestrict.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        with:
+          python-version: 3.8
+          architecture: x64
+
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
+
+      - name: Install Python Packages
+        run: |
+          pip3 install rockset==0.8.10
```
Comment (seemethere) [inline .github/workflows/update-viablestrict.yml:27]: One thing we might want to do in the future is consolidate these dependencies into flat files, perhaps in something like `.github/dependencies/update-viablestrict.txt` and then just use `pip install -r .github/dependencies/update-viablestrict.txt`.

That way we have a singular place to look if we need to update dependencies like this.

Not saying we have to do this now but something to think about in the future.

**Review by seemethere — APPROVED** [review]
Overall LGTM, just one nit / future idea

**Comment by malfet** [comment]
We should probably use something like [canary](https://github.com/pytorch/pytorch-canary) repo to test changes like that

**Pull request closed without merging**